H/HDF5/build_tarballs.jl

```diff
@@ -286,12 +298,13 @@ products = [
 # Dependencies that must be installed before this package can be built
 dependencies = [
     # To ensure that the correct version of libgfortran is found at runtime
-    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
-    Dependency("LibCURL_jll"; compat="7.73,8"),
-    Dependency("OpenSSL_jll"; compat="3.0.8"),
-    Dependency("Zlib_jll"),
-    # Dependency("dlfcn_win32_jll"; platforms=filter(Sys.iswindows, platforms)),
-    Dependency("libaec_jll"),   # This is the successor of szlib
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"), v"1.1.1";
+               compat="1.1"),
+    Dependency("LibCURL_jll", v"8.7.1"; compat="7, 8"),
```

Comment:
I always find odd using build version _and_ compat bound, especially if the build version isn't the first compatible one. Also, for libcurl see  #8407 
```suggestion
    Dependency("LibCURL_jll"; compat="7.73.0, 8"),
```

Reply:
It's difficult to set a good compat bound because the build will fail unless the specified version is one that can be downloaded. It's not easy to see which versions are available.